#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/robots/sphero/bb8/protos/BB-8.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/robots/softbank/nao/protos/Nao.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/robots/gctronic/e-puck/protos/E-puck.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/humans/pedestrian/protos/Pedestrian.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/factory/containers/protos/WoodenBox.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/robots/gctronic/e-puck/protos/E-puckGroundSensors.proto"

WorldInfo {
  basicTimeStep 16
  coordinateSystem "NUE"
  contactProperties [
    ContactProperties {
      coulombFriction [
        0
      ]
    }
  ]
}
Viewpoint {
  orientation -0.06419889632428742 0.6543505097125479 0.7534612877578197 3.866161110248018
  position 0.8572909907700326 1.7720813717997244 -0.3891009562394378
  follow "rectangle arena"
}
TexturedBackground {
  luminosity 0
}
TexturedBackgroundLight {
  texture "music_hall"
  luminosity 0
}
Solid {
  translation 0.173504 0 0.307859
  rotation 1 0 0 -1.5707996938995747
  children [
    Solid {
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/textures/parquetry/chequered_parquetry_base_color.jpg"
              ]
            }
            roughness 1
            roughnessMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/textures/parquetry/chequered_parquetry_roughness.jpg"
              ]
            }
            metalness 0
            normalMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/textures/parquetry/chequered_parquetry_normal.jpg"
              ]
            }
            occlusionMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/textures/parquetry/chequered_parquetry_occlusion.jpg"
              ]
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                -5 -5 0
                5 -5 0
                -5 5 0
                5 5 0
              ]
            }
            texCoord TextureCoordinate {
              point [
                0 0
                20 0
                0 20
                20 20
              ]
            }
            coordIndex [
              2, 0, 1, 3, -1
            ]
            texCoordIndex [
              2, 0, 1, 3, -1
            ]
          }
        }
      ]
      name "floor"
      model "floor"
      boundingObject Plane {
        size 10 10
      }
      locked TRUE
    }
    Solid {
      translation 5.005 0 0.05
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_base_color.jpg"
              ]
            }
            roughnessMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_roughness.jpg"
              ]
            }
            normalMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_normal.jpg"
              ]
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                -0.005 5 0.05
                0.005 5 0.05
                0.005 5 -0.05
                -0.005 5 -0.05
                -0.005 -5 -0.05
                0.005 -5 -0.05
                0.005 -5 0.05
                -0.005 -5 0.05
                -0.005 -5 0.05
                0.005 -5 0.05
                0.005 5 0.05
                -0.005 5 0.05
                0.005 -5 -0.05
                -0.005 -5 -0.05
                -0.005 5 -0.05
                0.005 5 -0.05
                -0.005 -5 -0.05
                -0.005 -5 0.05
                -0.005 5 0.05
                -0.005 5 -0.05
                0.005 -5 0.05
                0.005 -5 -0.05
                0.005 5 -0.05
                0.005 5 0.05
              ]
            }
            texCoord TextureCoordinate {
              point [
                0.001 0.01
                0 0.01
                0 0
                0.001 0
                0 0
                0.001 0
                0.001 0.01
                0 0.01
                0 0.001
                0 0
                1 0
                1 0.001
                0 0.001
                0 0
                1 0
                1 0.001
                1 0
                1 0.01
                0 0.01
                0 0
                0 0.01
                0 0
                1 0
                1 0.01
              ]
            }
            coordIndex [
              0, 1, 2, 3, -1, 4, 5, 6, 7, -1, 8, 9, 10, 11, -1, 12, 13, 14, 15, -1, 16, 17, 18, 19, -1, 20, 21, 22, 23, -1
            ]
            creaseAngle 1.5
          }
        }
      ]
      name "wall"
      model "box"
      boundingObject Box {
        size 0.01 10 0.1
      }
    }
    Solid {
      translation -5.005 0 0.05
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_base_color.jpg"
              ]
            }
            roughnessMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_roughness.jpg"
              ]
            }
            normalMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_normal.jpg"
              ]
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                -0.005 5 0.05
                0.005 5 0.05
                0.005 5 -0.05
                -0.005 5 -0.05
                -0.005 -5 -0.05
                0.005 -5 -0.05
                0.005 -5 0.05
                -0.005 -5 0.05
                -0.005 -5 0.05
                0.005 -5 0.05
                0.005 5 0.05
                -0.005 5 0.05
                0.005 -5 -0.05
                -0.005 -5 -0.05
                -0.005 5 -0.05
                0.005 5 -0.05
                -0.005 -5 -0.05
                -0.005 -5 0.05
                -0.005 5 0.05
                -0.005 5 -0.05
                0.005 -5 0.05
                0.005 -5 -0.05
                0.005 5 -0.05
                0.005 5 0.05
              ]
            }
            texCoord TextureCoordinate {
              point [
                0.001 0.01
                0 0.01
                0 0
                0.001 0
                0 0
                0.001 0
                0.001 0.01
                0 0.01
                0 0.001
                0 0
                1 0
                1 0.001
                0 0.001
                0 0
                1 0
                1 0.001
                1 0
                1 0.01
                0 0.01
                0 0
                0 0.01
                0 0
                1 0
                1 0.01
              ]
            }
            coordIndex [
              0, 1, 2, 3, -1, 4, 5, 6, 7, -1, 8, 9, 10, 11, -1, 12, 13, 14, 15, -1, 16, 17, 18, 19, -1, 20, 21, 22, 23, -1
            ]
            creaseAngle 1.5
          }
        }
      ]
      name "wall#2"
      model "box"
      boundingObject Box {
        size 0.01 10 0.1
      }
    }
    Solid {
      translation 0 5.005 0.05
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_base_color.jpg"
              ]
            }
            roughnessMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_roughness.jpg"
              ]
            }
            normalMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_normal.jpg"
              ]
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                -5.01 0.005 0.05
                5.01 0.005 0.05
                5.01 0.005 -0.05
                -5.01 0.005 -0.05
                -5.01 -0.005 -0.05
                5.01 -0.005 -0.05
                5.01 -0.005 0.05
                -5.01 -0.005 0.05
                -5.01 -0.005 0.05
                5.01 -0.005 0.05
                5.01 0.005 0.05
                -5.01 0.005 0.05
                5.01 -0.005 -0.05
                -5.01 -0.005 -0.05
                -5.01 0.005 -0.05
                5.01 0.005 -0.05
                -5.01 -0.005 -0.05
                -5.01 -0.005 0.05
                -5.01 0.005 0.05
                -5.01 0.005 -0.05
                5.01 -0.005 0.05
                5.01 -0.005 -0.05
                5.01 0.005 -0.05
                5.01 0.005 0.05
              ]
            }
            texCoord TextureCoordinate {
              point [
                1 0.009980039920159682
                0 0.009980039920159682
                0 0
                1 0
                0 0
                1 0
                1 0.009980039920159682
                0 0.009980039920159682
                0 1
                0 0
                0.0009980039920159682 0
                0.0009980039920159682 1
                0 1
                0 0
                0.0009980039920159682 0
                0.0009980039920159682 1
                0.0009980039920159682 0
                0.0009980039920159682 0.009980039920159682
                0 0.009980039920159682
                0 0
                0 0.009980039920159682
                0 0
                0.0009980039920159682 0
                0.0009980039920159682 0.009980039920159682
              ]
            }
            coordIndex [
              0, 1, 2, 3, -1, 4, 5, 6, 7, -1, 8, 9, 10, 11, -1, 12, 13, 14, 15, -1, 16, 17, 18, 19, -1, 20, 21, 22, 23, -1
            ]
            creaseAngle 1.5
          }
        }
      ]
      name "wall#3"
      model "box"
      boundingObject Box {
        size 10.02 0.01 0.1
      }
    }
    Solid {
      translation 0 -5.005 0.05
      children [
        Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_base_color.jpg"
              ]
            }
            roughnessMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_roughness.jpg"
              ]
            }
            normalMap ImageTexture {
              url [
                "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/textures/brushed_aluminium/brushed_aluminium_normal.jpg"
              ]
            }
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                -5.01 0.005 0.05
                5.01 0.005 0.05
                5.01 0.005 -0.05
                -5.01 0.005 -0.05
                -5.01 -0.005 -0.05
                5.01 -0.005 -0.05
                5.01 -0.005 0.05
                -5.01 -0.005 0.05
                -5.01 -0.005 0.05
                5.01 -0.005 0.05
                5.01 0.005 0.05
                -5.01 0.005 0.05
                5.01 -0.005 -0.05
                -5.01 -0.005 -0.05
                -5.01 0.005 -0.05
                5.01 0.005 -0.05
                -5.01 -0.005 -0.05
                -5.01 -0.005 0.05
                -5.01 0.005 0.05
                -5.01 0.005 -0.05
                5.01 -0.005 0.05
                5.01 -0.005 -0.05
                5.01 0.005 -0.05
                5.01 0.005 0.05
              ]
            }
            texCoord TextureCoordinate {
              point [
                1 0.009980039920159682
                0 0.009980039920159682
                0 0
                1 0
                0 0
                1 0
                1 0.009980039920159682
                0 0.009980039920159682
                0 1
                0 0
                0.0009980039920159682 0
                0.0009980039920159682 1
                0 1
                0 0
                0.0009980039920159682 0
                0.0009980039920159682 1
                0.0009980039920159682 0
                0.0009980039920159682 0.009980039920159682
                0 0.009980039920159682
                0 0
                0 0.009980039920159682
                0 0
                0.0009980039920159682 0
                0.0009980039920159682 0.009980039920159682
              ]
            }
            coordIndex [
              0, 1, 2, 3, -1, 4, 5, 6, 7, -1, 8, 9, 10, 11, -1, 12, 13, 14, 15, -1, 16, 17, 18, 19, -1, 20, 21, 22, 23, -1
            ]
            creaseAngle 1.5
          }
        }
      ]
      name "wall#4"
      model "box"
      boundingObject Box {
        size 10.02 0.01 0.1
      }
    }
  ]
  name "rectangle arena"
  model "rectangle arena"
}
Robot {
  translation 0.370854 0.67 -1.47926
  rotation 1 0 0 -1.5707953071795862
  children [
    Camera {
    }
    E-puck {
      translation 0 -1.255674059841283e-06 0
      rotation 0.9999999999946791 3.2621478573824188e-06 6.50178572327721e-11 3.9239998069237915e-10
    }
  ]
  name "robot(1)"
  boundingObject E-puckGroundSensors {
  }
  physics Physics {
  }
  selfCollision TRUE
}
Transform {
  translation 1.29 0 -3.61
  children [
    Solid {
      translation 0 1.6484162134853948 1.5636182664202636e-06
      rotation -1 3.4859479231043136e-17 -1.9032793650619137e-17 1.5707986954477837
      children [
        SpotLight {
          attenuation 0 0 1
          beamWidth 0.7
          color 0.666667 0 0
          cutOffAngle 0.78538
          direction 44 -77 -9
          intensity 5
          location 0 3.1 2
        }
        Shape {
          appearance PBRAppearance {
          }
          geometry Cylinder {
            height 3.1
            radius 0.11
          }
        }
      ]
      name "poste"
      boundingObject Cylinder {
        height 3.1
        radius 0.11
      }
      physics Physics {
        mass 2
      }
      locked TRUE
    }
    Solid {
      translation 0 0.04861342874229697 -1.4444730656038018e-07
      rotation -1 -8.465995838554186e-18 6.474439145104106e-19 1.5707995869678204
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Cylinder {
            height 0.1
            radius 0.5
          }
        }
      ]
      name "base"
      boundingObject Cylinder {
        height 0.1
        radius 0.5
      }
      physics Physics {
        mass 5
      }
      locked TRUE
    }
  ]
}
WoodenBox {
  translation -0.4376810000000001 0.2996048471796917 -0.5224730047134802
  rotation -1 8.918301052746881e-12 1.405111730333671e-13 3.3708718369325186e-06
  mass 2
}
WoodenBox {
  translation 2.18 0.299605360465655 -0.37000200471520844
  rotation -1 1.0069648843013658e-11 -1.6863279306077027e-12 3.3708718369301435e-06
  name "wooden box(1)"
  mass 2
}
WoodenBox {
  translation 2.95 0.2996007820939639 -1.7300020046997922
  rotation -1 9.751233172143009e-12 8.43140944394786e-13 3.3708718369329857e-06
  name "wooden box(2)"
  mass 2
}
Robot {
  translation 0 0.96 0
  children [
    SliderJoint {
      jointParameters JointParameters {
      }
      endPoint Solid {
        children [
          HingeJoint {
            endPoint Solid {
              children [
                Camera {
                }
              ]
            }
          }
        ]
      }
    }
    Camera {
      name "CAM"
    }
  ]
  controller "<none>"
  controllerArgs [
    "wb_camera_enable"
  ]
  supervisor TRUE
}
DEF BB-8 BB-8 {
  translation 0.3289045319575719 -0.020500776132553453 1.1202104209419486
  rotation -0.905887565081058 0.2929067462481501 0.30589762574075985 1.7172366944983708
  controller "sumo_supervisor"
}
Solid {
  translation 0 -7.348239359999997 0
  physics Physics {
  }
}
Robot {
  name "supervisor"
  controller ""
  supervisor TRUE
}
DEF BALL Solid {
  translation 0.593641 0.19989295913338104 0.06433313167516444
  rotation 0.3619895332355768 -0.7583841949894695 0.5420488821297057 1.8709830763951871
  children [
    Shape {
      appearance DEF SPHERE_COLOR PBRAppearance {
        baseColor 1 0 0
      }
      geometry DEF SPHERE_GEOMETRY Sphere {
        radius 0.2
      }
    }
  ]
  name "solid(1)"
  boundingObject USE SPHERE_GEOMETRY
  physics Physics {
    density -1
    mass 0.1
  }
}
Nao {
  translation -2.002635820010987 0.33327304893911563 0.9301091625589591
  rotation -0.6021538009701056 0.5645782400010415 0.5645017368392952 2.05769496627077
  controller "nao_by_ai"
  supervisor TRUE
  cameraFocus Focus {
  }
}
Pedestrian {
  translation -2.79224 1.54 3.7
  rotation -0.9351134888498739 0.25056313098704747 0.2505611309860019 1.63784
}
Solid {
  translation 7.83 8.87 9.26
  children [
    PointLight {
      attenuation 0 0 1
      intensity 50
      location 0 0.3 0
      castShadows TRUE
    }
    Shape {
      geometry Sphere {
      }
    }
  ]
  name "sun"
}
Solid {
  translation 0 1.1 0
  rotation 0 1 0 1.5708
  children [
    Group {
      children [
        Transform {
          translation 0 0 0.05
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Sphere {
                radius 0.03
                subdivision 2
              }
            }
          ]
        }
        Transform {
          translation 0 0 -0.04
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Sphere {
                radius 0.03
                subdivision 2
              }
            }
          ]
        }
        Shape {
          appearance PBRAppearance {
          }
          geometry Cylinder {
            height 0.1
            radius 0.01
          }
        }
      ]
    }
  ]
  name "pesas"
  contactMaterial "dumbbell"
  boundingObject USE SPHERE_GEOMETRY
  physics Physics {
    density -1
    mass 2
    centerOfMass [
      0 0 0.01
    ]
  }
}
